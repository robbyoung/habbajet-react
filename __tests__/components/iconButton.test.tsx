import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from '../../app/components/iconButton';
import {faCoffee} from '@fortawesome/free-solid-svg-icons';
import {render, fireEvent, wait} from '@testing-library/react-native';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

describe('IconButton Component', () => {
    it('will render a button with a font-awesome icon', () => {
        const component = renderer.create(
            <IconButton
                icon={faCoffee}
                size={70}
                color="#036bfc"
                onPress={() => undefined}
            />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('will run the onPress callback if the button is pressed', async () => {
        const onPress = jest.fn();
        const onLongPress = jest.fn();

        const {getByTestId} = render(
            <IconButton
                icon={faCoffee}
                size={70}
                color="#036bfc"
                testID="button-icon"
                onPress={onPress}
                onLongPress={onLongPress}
            />,
        );

        const button = getByTestId('button-icon');
        fireEvent.press(button);
        await wait(() => {
            expect(onPress).toBeCalledTimes(1);
            expect(onLongPress).toBeCalledTimes(0);
        });
    });

    it('will run the onLongPress callback if the button is long pressed', async () => {
        const onPress = jest.fn();
        const onLongPress = jest.fn();
        const {getByTestId} = render(
            <IconButton
                icon={faCoffee}
                size={70}
                color="#036bfc"
                testID="button-icon"
                onPress={onPress}
                onLongPress={onLongPress}
            />,
        );

        const arrowButton = getByTestId('button-icon');
        fireEvent.longPress(arrowButton);
        await wait(() => {
            expect(onPress).toBeCalledTimes(0);
            expect(onLongPress).toBeCalledTimes(1);
        });
    });
});
