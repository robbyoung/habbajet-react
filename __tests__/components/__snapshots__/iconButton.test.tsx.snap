// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Component will render a button with a font-awesome icon 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <
    color="#036bfc"
    icon={
      Object {
        "icon": Array [
          640,
          512,
          Array [],
          "f0f4",
          "M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z",
        ],
        "iconName": "coffee",
        "prefix": "fas",
      }
    }
    size={70}
  />
</View>
`;
